### PR TITLE
FEAT(stark252 field): Adds Stark252 curve

### PR DESCRIPTION
**Pull request by PatStiles**
## Describe the changes

Rebase of #493 after merging of V2.

### CODE REVIEW

**Comment by jeremyfelder** [comment]
@PatStiles Thanks for adding this! 🚀 👍🏻 
Can you format the Rust and Cpp code according to https://github.com/ingonyama-zk/icicle?tab=readme-ov-file#development-contributions

**Comment by PatStiles** [comment]
@jeremyfelder I'm having trouble addressing the failing cargo test in the CI which is failing due to a lack of ArkField  declarationfor stark252 (arkworks does not implement it). Out of curiosity I ran the same test for icicle-babybear and saw it also failed within that crate. Let me know if you have any suggestions for addressing the issue.
Screenshot 2024-04-25 at 15 05 21

**Comment by jeremyfelder** [comment]
> @jeremyfelder I'm having trouble addressing the failing cargo test in the CI which is failing due to a lack of ArkField declarationfor stark252 (arkworks does not implement it). Out of curiosity I ran the same test for icicle-babybear and saw it also failed within that crate. Let me know if you have any suggestions for addressing the issue. Screenshot 2024-04-25 at 15 05 21

The CI tests for stark252 need to be separated out like the babybear tests; https://github.com/PatStiles/icicle/blob/feat/stark252_field/.github/workflows/rust.yml#L64-L73.

**Review by jeremyfelder — CHANGES_REQUESTED** [review]
@PatStiles Looks great!
Can you add a build.rs file for stark252 to link to the static lib. Similar to babybear's but with the EXT_FIELD = OFF

**Comment by PatStiles** [comment]
@jeremyfelder addressed!

File: scripts/gen_c_api.py [hunk 1]
```diff
@@ -53,6 +53,10 @@
 FIELDS_CONFIG = {
     "babybear": {
         "poseidon.h",
+    },
+    "stark252": {
+        "poseidon.h",
+        "field_ext.h",
```
[the remark this answers is not in the document]
Comment (ChickenLover) [reply scripts/gen_c_api.py:59]: yes, maybe we should modify this script to be more intuitive. @PatStiles can you add these?

[the remark this answers is not in the document]
Comment (ChickenLover) [reply scripts/gen_c_api.py:59]: don't forget to re-run the script after modifying

File: wrappers/rust/icicle-fields/icicle-stark252/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "icicle-stark252"
+version.workspace = true
+edition.workspace = true
+authors.workspace = true
+description = "Rust wrapper for the CUDA implementation of baby bear prime field by Ingonyama"
```
Comment (ChickenLover) [inline wrappers/rust/icicle-fields/icicle-stark252/Cargo.toml:6]: baby bear -> stark252

**Comment by PatStiles** [comment]
@ChickenLover Should be resolved!

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -60,10 +60,10 @@ jobs:
       if: needs.check-changed-files.outputs.rust == 'true' || needs.check-changed-files.outputs.cpp_cuda == 'true'
       # Running tests from the root workspace will run all workspace members' tests by default
       # We need to limit the number of threads to avoid running out of memory on weaker machines
-      # ignored tests are polynomial tests. Since they conflict with NTT tests, they are executed sperately
+      # ignored tests are polynomial tests. Since they conflict with NTT tests, they are executed separately
       run: |
-        cargo test --workspace --exclude icicle-babybear --release --verbose --features=g2 -- --test-threads=2 --ignored
-        cargo test --workspace --exclude icicle-babybear --release --verbose --features=g2 -- --test-threads=2
+        cargo test --workspace --exclude icicle-babybear icicle-stark252 --release --verbose --features=g2 -- --test-threads=2 --ignored
```
Comment (jeremyfelder) [inline .github/workflows/rust.yml:65]: The rust docs are not very explicit about this but I think the `--exclude` flag should be specified per package or glob pattern.

```suggestion
        cargo test --workspace --exclude icicle-babybear --exclude icicle-stark252 --release --verbose --features=g2 -- --test-threads=2 --ignored
```

Same change for the line following

**Review by jeremyfelder — CHANGES_REQUESTED** [review]
I think this should be the last change

File: .github/workflows/rust.yml (hunk 3 above)
Comment (PatStiles) [reply .github/workflows/rust.yml]: addressed!

**Review by jeremyfelder — APPROVED** [review]

**Review by LeonHibnik — APPROVED** [review]

**Review by ChickenLover — APPROVED** [review]

**Pull request merged**